### PR TITLE
Run eslint on travis

### PR DESCRIPTION
**Pull request by simfeld**
See #657

### CODE REVIEW

**Comment by simfeld** [comment]
Options:

1. Lint in the test stage (as done currently)
2. Have a separate linting stage which runs `npm run lint` and `composer cs-check`, failing the build early if there are errors
3. Something else?

File: frontend/src/components/camp/StoryDay.vue [hunk 1]
```diff
@@ -80,6 +80,7 @@ export default {
         }
       })
     },
+    Houston calling linter
```
Comment (carlobeltrame) [inline frontend/src/components/camp/StoryDay.vue]: :rofl:

**Comment by carlobeltrame** [comment]
I'd go for Option 2, and also lint the print service while we're at it

File: .travis.yml [hunk 2]
```diff
@@ -5,6 +5,51 @@ notifications:
...
+      stage: lint
+      language: php
+      php:
+        - 7.4.2
+      cache:
+        directories:
+          - vendor
+      before_script:
+          - cd backend
+          - composer install --no-interaction
```
Comment (simfeld) [inline .travis.yml:18]: I guess we wouldn't need to install all the dependencies here, but we are caching `/backend/vendor` anyway.

File: .travis.yml [hunk 3]
```diff
@@ -5,6 +5,51 @@ notifications:
...
+          - cd backend
+          - composer cs-check
+          - cd ..
+
+    - name: "Frontend lint"
+      stage: lint
+      language: node_js
+      node_js: '10'
+      env:
+        - NODE_ENV=development
```
Comment (simfeld) [inline .travis.yml:30]: Not entirely sure whether this is needed here.

File: print/static/print-preview.css [hunk 4]
```diff
@@ -66,6 +66,7 @@
 
     /* uncomment this part for recto/verso book : ------------------------------------ */
 
+    /* stylelint-disable no-duplicate-selectors, no-descending-specificity */
```
Comment (simfeld) [inline print/static/print-preview.css:69]: @usu had to disable some rules here to make stylelint happy, I didn't want to tinker with your CSS 😉

**Review by simfeld — COMMENTED** [review]
Overall build time increased by ~3mins because of the separate lint stage, compare https://travis-ci.com/github/ecamp/ecamp3/pull_requests.

File: .travis.yml (hunk 2 above)
Comment (simfeld) [reply .travis.yml:18]: Or maybe we aren't, as the cache path is only `vendor`? And maybe we shouldn't even, after reading this https://github.com/travis-ci/travis-ci/issues/4579? Another post that suggests caching `$HOME/.composer/cache/files`: https://blog.wyrihaximus.net/2015/07/composer-cache-on-travis/.

Comment (carlobeltrame) [reply .travis.yml:18]: True, but we can handle correct caching in #681. So this is fine with me for now.

**Review by carlobeltrame — APPROVED** [review]

**Review by usu — APPROVED** [review]

**Pull request merged**